lib/vrl/parser/src/template_string.rs

```diff
@@ -0,0 +1,76 @@
+use diagnostic::Span;
+use std::fmt;
+
+use crate::ast::{Expr, Ident, Literal::RawString, Node, Op, Opcode};
+
+#[derive(Clone, PartialEq, Eq, Ord, PartialOrd, Debug, Hash)]
+pub struct TemplateString(pub Vec<StringSegment>);
+
+impl TemplateString {
+    /// Rewrites the ast for the template string to be a series of string concatenations
+    pub fn rewrite(&self) -> Expr {
+        self.0
+            .iter()
+            .map(|node| -> Expr {
+                match node {
```

Comment:
nit

```suggestion
            .map(|node| match node {
```

You'll also need to remove a single `}` at the end of this block.